sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSComposeApp.kt

```diff
@@ -209,15 +216,115 @@ fun POSComposeApp(
+                            amount = snapAmount,
+                            description = "Testing POS certification app payments (SNAP Purchase)",
+                            fundingType = FundingType.EBTSnap.value,
+                            paymentMethodRef = "",
+                            posTerminal = PosTerminal(providerTerminalId = k9SDK.terminalId),
+                            metadata = mapOf()
+                        )
+                        viewModel.setLocalPayment(payment = payment)
+                        navController.navigate(POSScreen.PAYChoosePANMethodScreen.name)
+                    },
+                    onCancelButtonClicked = { navController.popBackStack(POSScreen.PAYTransactionTypeSelectionScreen.name, inclusive = false) }
+                )
+            }
+            composable(route = POSScreen.PAYChoosePANMethodScreen.name) {
+                PANMethodSelectionScreen(
+                    onManualEntryButtonClicked = { navController.navigate(POSScreen.PAYManualPANEntryScreen.name) },
+                    onSwipeButtonClicked = { navController.navigate(POSScreen.PAYMagSwipePANEntryScreen.name) },
+                    onBackButtonClicked = { navController.popBackStack(POSScreen.PAYTransactionTypeSelectionScreen.name, inclusive = false) }
+                )
+            }
+            composable(route = POSScreen.PAYManualPANEntryScreen.name) {
+                ManualPANEntryScreen(
+                    forageConfig = uiState.forageConfig,
+                    onSubmitButtonClicked = {
+                        Log.i("POSComposeApp", "Calling onSubmitButtonClicked in ManualPANEntryScreen in PAYChoosePANMethodScreen")
+
+                        if (panElement != null) {
+                            viewModel.tokenizeEBTCard(
+                                panElement as ForagePANEditText,
+                                k9SDK.terminalId,
+                                onSuccess = { tokenizedCard ->
+                                    Log.i("POSComposeApp", "payment method? — $tokenizedCard")
+                                    if (tokenizedCard?.ref != null && uiState.localPayment != null) {
+                                        Log.i("POSComposeApp", "Successfully tokenized EBT card with ref: $tokenizedCard.ref")
+                                        val payment = uiState.localPayment!!.copy(paymentMethodRef = tokenizedCard.ref)
+                                        viewModel.createPayment(payment = payment, merchantId = uiState.merchantId, onSuccess = { serverPayment ->
+                                            if (serverPayment.ref !== null) {
+                                                navController.navigate(POSScreen.PAYPINEntryScreen.name)
+                                            }
+                                        })
+                                    }
+                                }
+                            )
+                        }
```

Comment:
This is kinda gross and can probably be improved, especially since it's duplicated in the swipe version

Reply:
not a show stopper from my pov! Feel free to clean this code up whenever but I could see a natural checkpoint to be after we have all the Payment flows working (Tiffany and Finn will be unblocked at that point)